Sitemap/Url/GoogleVideo.php

```diff
@@ -182,54 +182,263 @@ class GoogleVideo
+                    $this->setCategory($param);
+                    break;
+                case 'restriction_allow':
+                    $this->setRestrictionAllow($param);
+                    break;
+                case 'restriction_deny':
+                    $this->setRestrictionDeny($param);
+                    break;
+                case 'gallery_loc':
+                    @trigger_error(
+                        'GoogleVideo parameter ' . $key . ' is deprecated since 2.3.0, use "gallery_location" parameter instead',
+                        E_USER_DEPRECATED
+                    );
+                    $this->setGalleryLocation($param);
+                    break;
+                case 'gallery_location':
+                    $this->setGalleryLocation($param);
+                    break;
+                case 'gallery_loc_title':
+                    @trigger_error(
+                        'GoogleVideo parameter ' . $key . ' is deprecated since 2.3.0, use "gallery_location_title" parameter instead',
+                        E_USER_DEPRECATED
+                    );
+                    $this->setGalleryLocationTitle($param);
+                    break;
+                case 'gallery_location_title':
+                    $this->setGalleryLocationTitle($param);
+                    break;
+                case 'requires_subscription':
+                    $this->setRequiresSubscription($param);
+                    break;
+                case 'uploader':
+                    $this->setUploader($param);
+                    break;
+                case 'uploader_info':
+                    $this->setUploaderInfo($param);
+                    break;
+                case 'platforms':
+                    $this->setPlatforms($param);
+                    break;
+                case 'platform_relationship':
+                    $this->setPlatformRelationship($param);
+                    break;
+                case 'live':
+                    $this->setLive($param);
+                    break;
+            }
         }
 
-        $this->setThumbnailLoc($thumnail_loc);
+        $this->setThumbnailLocation($thumbnailLocation);
         $this->setTitle($title);
         $this->setDescription($description);
 
-        if (!$this->content_loc && !$this->player_loc) {
-            throw new Exception\GoogleVideoException('The parameter content_loc or player_loc is required');
+        if (!$this->contentLocation && !$this->playerLocation) {
+            throw new Exception\GoogleVideoException('The parameter content_location or content_location is required');
         }
 
-        if (count($this->platforms) && !$this->platform_relationship) {
+        if (count($this->platforms) && !$this->platformRelationship) {
             throw new Exception\GoogleVideoException(
                 'The parameter platform_relationship is required when platform is set'
             );
         }
     }
 
+    public function __get(string $name)
```

Comment:
Same questions: why do you use a typed property and why adding these magic methods?

Reply:
Same input : same output 🙂